test/ReactWrapper-spec.js

```diff
@@ -429,15 +429,24 @@ describeWithDOM('mount', () => {
     it('should support boolean and numeric values for matching props', () => {
       const wrapper = mount(
         <div>
-          <span value={1} />
+          <a value={0} />
+          <a value={1} />
           <a value={false} />
+          <a value />
         </div>
       );
 
-      expect(wrapper.find('span[value=1]')).to.have.length(1);
-      expect(wrapper.find('span[value=2]')).to.have.length(0);
-      expect(wrapper.find('a[value=false]')).to.have.length(1);
-      expect(wrapper.find('a[value=true]')).to.have.length(0);
+      expect(wrapper.find('[value=0]')).to.have.length(1);
+      expect(wrapper.find('[value=1]')).to.have.length(1);
+      expect(wrapper.find('[value=2]')).to.have.length(0);
+
+      expect(wrapper.find('[value=false]')).to.have.length(1);
+      expect(wrapper.find('[value=true]')).to.have.length(1);
+
+      expect(wrapper.find('[value="0"]')).to.have.length(0);
+      expect(wrapper.find('[value="1"]')).to.have.length(0);
+      expect(wrapper.find('[value="false"]')).to.have.length(0);
+      expect(wrapper.find('[value="true"]')).to.have.length(0);
```

Comment:
I specifically want to check for string selectors, since that's what's affected by this PR.